### PR TITLE
ROC-7204: Handle other court selection

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7204

### Change description ###
Handle other court selection on JDDO

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/HearingCourtDetailsFinder.java [hunk 1]
```diff
@@ -14,7 +20,18 @@ public HearingCourtDetailsFinder(CourtFinderApi courtFinderApi, HearingCourtMapp
         this.hearingCourtMapper = hearingCourtMapper;
     }
 
-    public HearingCourt findHearingCourtAddress(CCDHearingCourtType courtType) {
+    public HearingCourt findHearingCourtAddress(CCDCase ccdCase) {
+        return Optional.ofNullable(ccdCase.getHearingCourt())
+            .filter(c -> c != OTHER)
+            .map(this::mapHearingCourt)
+            .orElse(
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/HearingCourtDetailsFinder.java]: Prefer `orElseGet(Supplier)` where possible

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4956693&bid=16380375)

**Pull request closed without merging**